### PR TITLE
Add sentry

### LINKED ISSUE #82: Add and configure Sentry
**Is your feature request related to a problem? Please describe.**
We currently lack visibility into errors that occur on our site and rely on the goodwill of others to report these errors.  There are tools on the market that will automatically capture errors while providing enhanced workflows for resolving the bugs.  My favorite among these tools is [Sentry.io](https://sentry.io/welcome/); I've been using them for years with great success.

Sentry offers open-source sponsorship and has accepted our request for sponsorship because they. are. awesome.

The only thing left to do is actually install and configure their tool.  

---

If anyone has suggestions to offer, or questions before Sentry is installed please comment below.

### PR DESCRIPTION
**Pull request by brentmclark**
Closes #82 

Add, initialize and configure sentry with user info whenever it's present.

You can get the sentry `DSN` from the settings on their website or I can send it in DM.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -2,6 +2,10 @@ import * as dotenv from 'dotenv';
 import * as fs from 'fs';
 dotenv.config();
 
+import Config from './config';
+import * as Sentry from '@sentry/node';
+Sentry.init({ dsn: Config.sentry.DSN });
```
Comment (crysfel) [inline src/main.ts]: Just wondering how are we going to handle the data from different environments, are we going to have a different `DSN` value for each environment?

File: src/modules/users/users.controller.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import * as Sentry from '@sentry/node';
```
Comment (crysfel) [inline src/modules/users/users.controller.ts]: Do we need to import sentry in all other controllers?

File: src/main.ts (hunk 1 above)
Comment (brentmclark) [reply src/main.ts]: Nope, same DSN for each environment.  It's pretty easy to tag each environment with an environment variable upon initialization of Sentry which can then be used as a filter in the Sentry UI.  I'm happy to hook that up if you can guide me toward how the code knows which environment it's in.

It's also worth mentioning that a missing DSN is the correct way to tell Sentry not to log anything, so it's perfectly fine for it to remain blank on our local environments.

File: src/modules/users/users.controller.ts (hunk 2 above)
Comment (brentmclark) [reply src/modules/users/users.controller.ts]: I don't think so; only where we need to use its API directly; for example setting tags, configuring the user, or calling sentry explicitly.

**Comment by brentmclark** [comment]
@crysfel Are we good to merge this?

**Comment by crysfel** [comment]
Yes! Thanks @brentmclark

**Pull request merged**